portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,30 +50,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     int requestLimit = Integer.parseInt(request.getParameter("request-limit"));
     String language = request.getParameter("language");
 
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-
-    List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(requestLimit))) {
-      String author = (String) entity.getProperty("author");
-      String text = (String) entity.getProperty("text");
-
-      Translate translate = TranslateOptions.newBuilder().setProjectId("internplayground")
-          .setQuotaProjectId("internplayground").build().getService();
-      Translation translation = translate.translate(text,
-          Translate.TranslateOption.targetLanguage(language));
-      text = translation.getTranslatedText();
-      Comment comment = new Comment(author, text);
-      comments.add(comment);
+    Translate translate = TranslateOptions.newBuilder().setProjectId("internplayground")
+        .setQuotaProjectId("internplayground").build().getService();
+
+    List<Language> languages = translate.listSupportedLanguages();
+    
+    Boolean langParamValid = false;
+    for (Language lang : languages) {
+      if (lang.getCode().equals(language)) {
+        langParamValid = true;
+        break;
+      }
+    }
+    if (langParamValid) {
```

Comment:
Avoid large condition blocks; often the reverse logic can be used to return early, or perform some smaller task.

In this case, you could use the reverse logic to change the selected language to a default, bonus for being able to inject a warning/notice in the resulting page that an invalid language was detected.